content/docs/learn/typed-errors/from-either-to-raise.md

```diff
@@ -0,0 +1,260 @@
+computations that are successful or end in error. This style is
+fully supported by Arrow, but the DSL based around `Raise` usually
+results in nicer code. This small guide describes common patterns
+in the `Either` style and how they translate into `Raise`.
+
+:::note Working with typed errors
+
+For a more general introduction to typed errors, without assuming
+any prior knowledge on the topic, check [this tutorial](working-with-typed-errors.md).
+
+:::
+
+In the following discussion, we assume the following functions:
+
+```kotlin
+fun f(n: Int): Either<Error, String>
+fun g(s: String): Either<Error, Thing>
+fun h(s: String): Either<Boo, Thing>
+fun Thing.summarize(): String
+```
+
+## Sequential composition
+
+The basic way to compose computations that may error out is by
+sequentially executing each of them, and finishing early if
+an error appears. This is done using `flatMap` — if the next
+computation may also fail — or `map` — if you need to apply
+a pure computation to the result, if available.
+
+Take this possible combination of the aforementioned functions:
+
+```kotlin
+fun foo(n: Int): Either<Error, String> =
+  f(n).flatMap { s ->
+    g(s).map { t ->
+      t.summarize()
+    }
+  }
+```
+
+The translation into the `Raise` DSL describes the same sequence
+of steps, but using a more sequential style:
+
+```kotlin
+fun foo(n: Int): Either<Error, String> = either {
+  val s = f(n).bind()
+  val t = g(s).bind()
+  t.summarize()
+}
+```
+
+There are two important steps in this translation. At the beginning of
+the function, we have the `either` builder, which (1) indicates that
+the result of the function is `Either` of some error type and (2)
+creates a new scope in which you can use the `Raise` DSL.
+
+The second step is that every time that you have an `Either` value
+resulting from another computation (like `f` and `g` above), you need
+to call `bind`. This "injects" any potential error value in the context,
+and continues normal execution otherwise.
+In this style you don't need to use any function to compose computations,
+you use regular Kotlin idioms with some `bind`s sprinkled. In fact
+you could have written the previous code in one single line:
+
+```kotlin
+fun foo(n: Int): Either<Error, String> = either {
+  g(f(n).bind()).bind().summarize()
+}
+```
+
+How you split your code into local values no longer depends on the
+structure of your functions, as is the case with `flatMap`, but rather
+on the logical decomposition you want in your code.
+
+Arrow provides different builders for different return types
+(`either`, `option`, `result`), but regardless of the one you choose
+you always use `bind` for the injection phase.
+
+:::info Why "Raise DSL"?
+
+At this point, you may be surprised that we haven't used the word `Raise`
+at all in the code, only `either` and `bind`. To understand why we use
+"`Raise` DSL" to refer to this coding style, we need to dig a big in
```

Comment:
Did you mean "dig a bit into"? 

Also, possibly better to refer to it as "definition", "signature" or "declaration" of `either` for clarity. The "type" of `either` being referred to is assumedly the function-type of it's signature, i.e. `(Raise<E>.() -> A) -> Either<E, A>`, however some may instead think of the "type" of a function to be that of the return value.

Reply:
Agree, I've rephrased it slightly.